test/lib/rules/no-uppercase-with-underscore-property-name.js

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+'use strict';
+
+const { RuleTester } = require('eslint');
+
+const { ESLINT_TEST_CONFIG } = require('../shared');
+const rule = require('../../../lib/rules/no-uppercase-with-underscore-property-name');
+
+const ruleTester = new RuleTester(ESLINT_TEST_CONFIG);
+
+ruleTester.run('no-uppercase-with-underscore-property-name', rule, {
+    valid: [
```

Comment:
I would also add the following test to the list.

Valid cases:
- properties without uppercases or underscore: `foobar`

Invalid cases:
- properties starting with an uppercase containing an underscore: `Foo_bar`
- properties starting with an uppercase and finishing with an underscore: `Foobar_`
- properties starting with an underscore and finishing with an uppercase: `_foobaR`